### PR TITLE
Upgrade kubernetes-client-api to 6.x

### PR DESCRIPTION
**Pull request by Vlatombe**
Updates required kubernetes-client-api to 6.x

Downstream of
* [x] https://github.com/jenkinsci/kubernetes-client-api-plugin/pull/163
* [x] https://github.com/jenkinsci/kubernetes-credentials-plugin/pull/36

Task list
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -76,7 +76,7 @@
     <dependency>
       <groupId>org.jenkinsci.plugins</groupId>
       <artifactId>kubernetes-credentials</artifactId>
-      <version>0.9.0</version>
+      <version>0.9.1-rc136.6105f8f63f7f</version>
```
Comment (Vlatombe) [inline pom.xml:79]:
```suggestion
      <version>0.10.0</version>
```

File: pom.xml [hunk 2]
```diff
@@ -55,7 +55,7 @@
     <dependency>
       <groupId>org.jenkins-ci.plugins</groupId>
       <artifactId>kubernetes-client-api</artifactId>
-      <version>5.12.1-187.v577c3e368fb_6</version>
+      <version>6.3.1-205.vea_a_fc21de909</version>
```
Comment (Vlatombe) [inline pom.xml:58]:
```suggestion
      <version>6.3.1-206.v76d3b_6b_14db_b</version>
```

**Comment by s7an-it** [comment]
After this upgrade I am getting https://issues.jenkins.io/browse/JENKINS-70392?jql=project%20%3D%20JENKINS%20AND%20component%20%3D%20kubernetes-plugin, is there any tip how to avoid with custom build until there is official fix?

**Comment by fajarhide** [comment]
@Vlatombe i'm also issue after i'm update plugin the `kubernetes-client-api` version `5.12.2` to `6.3.1` get error 
```
hudson.remoting.ProxyException: io.fabric8.kubernetes.client.KubernetesClientException: No httpclient implementations found on the context classloader, please ensure your classpath includes an implementation jar
```

for currently solving I rolled back the version to `5.12.2` and it running well again.

please fix this soon

**Comment by jglick** [comment]
Fix coming in #1289 IIUC

**Comment by Vlatombe** [comment]
Fixed in [3800.v54b_077b_94b_a_4](https://github.com/jenkinsci/kubernetes-plugin/releases/tag/3800.v54b_077b_94b_a_4)

**Comment by jglick** [comment]
> not sure if it is right place to raise that issue

In general, file bug reports in Jira, though it is prudent to leave a mention in GitHub as well.

**Pull request merged**